contracts/ColonyNetwork.sol

```diff
@@ -120,9 +121,15 @@ contract ColonyNetwork is ColonyNetworkStorage {
     IColony colony = IColony(etherRouter);
     colony.setToken(_tokenAddress);
 
-    Authority authority = new Authority(colony);
+    ColonyAuthority colonyAuthority = new ColonyAuthority(colony);
+    Authority authority = new Authority(address(colonyAuthority));
```

Comment:
Why are you explicitly casting `ColonyAuthority` to `Authority` here. This is needed in JS but Solidity understands inheritance so there shouldn't be a need.

Reply:
So i'm creating authority for colony's authority and setting it [here](https://github.com/JoinColony/colonyNetwork/pull/224/files#diff-94738a1e114b0a8e909bb03e34628bf6R126), as you explained me. I'm just not sure how to name it, `ColonyAuthorityAuthority` sounds funny. 😄

Reply:
This part is confusing, so i will add some comments

Reply:
My note was about the cast, not about the name.

Reply:
The only cast i see here is `ColonyAuthority` to `address`, then passing it to `Authority`'s constructor

Reply:
Sorry I'm blind. Ignore me.